src/files-card-body.jsx

```diff
@@ -342,94 +351,95 @@ export const FilesCardBody = ({
-                        </Gallery>
-                    </CardBody>}
-                {!isGrid &&
-                    <ListingTable
-                      id="folder-view"
-                      className="pf-m-no-border-rows"
-                      variant="compact"
-                      columns={[_("Name")]}
-                      rows={sortedFiles.map(file => ({
-                          columns: [
-                              {
-                                  title: (
-                                      <Item
-                                        file={file}
-                                        key={file.name}
-                                        isSelected={!!selected.find(s => s.name === file.name)}
-                                        isGrid={isGrid}
-                                      />)
-                              }
-                          ]
-                      }))}
-                    />}
-            </div>
+            {sortedFiles.length !== 0 &&
+                <Table
+                  id="folder-view"
+                  className={`pf-m-no-border-rows fileview ${isGrid ? 'view-grid' : 'view-details'}`}
+                  variant="compact"
+                >
+                    <Thead>
+                        <Tr>
+                            <Th sort={sortColumn(0)} className="col-name">{_("Name")}</Th>
+                            <Th sort={sortColumn(1)} className="col-size">{_("Size")}</Th>
+                            <Th
+                              sort={sortColumn(2)} className="col-date"
+                              modifier="nowrap"
+                            >{_("Modified")}
+                            </Th>
+                        </Tr>
+                    </Thead>
+                    <Tbody>
+                        {sortedFiles.map((file, rowIndex) =>
+                            <Row
+                              key={rowIndex}
+                              file={file}
+                              isSelected={selected.some(s => s.name === file.name)}
+                            />)}
+                    </Tbody>
+                </Table>}
         </div>
     );
 };
 
-// Memoize the Item component as rendering thousands of them on each render of parent component is costly.
-const Item = React.memo(function Item({ file, isSelected, isGrid }) {
-    function getFileType(file) {
-        if (file.type === "dir") {
-            return "directory-item";
-        } else if (file.type === "lnk" && file?.to === "dir") {
-            return "directory-item";
-        } else {
-            return "file-item";
-        }
+const getFileType = (file) => {
+    if (file.type === "dir" || file.to === "dir") {
+        return "folder";
+    } else {
+        return "file";
     }
+};
+
+// Memoize the Item component as rendering thousands of them on each render of parent component is costly.
+const Row = React.memo(function Item({ file, isSelected }) {
```

Comment:
For the record, I don't think the memoize currently protects us from changes to unrelated files since we re-create the list on every single fsinfo event, but I'm happy to see what we are theoretically memoized, because it means that we can fix that bug and benefit from it...

Reply:
It does, I tested this on /bin, the event listener took ~ 200-300ms and with memozing its a lot faster.

React can be a bit confusing :)

So re-render => props/state changes and the render function is called again forcing all children to "re-render". By memozing React does not re-run the `render` function as the props did not change.

Reply:
Another issue might be that we memoize the sort order of the array. But then again changing our state to an Object might help with that :)